ezyrb/mapper.py

```diff
@@ -0,0 +1,243 @@
+"""
+Utilities for mapping solutions on a different grid
+"""
+import sys
+import numpy as np
+from ezyrb.filehandler import FileHandler
+import scipy
+
+
+class Mapper(object):
+	"""
+	Documentation
+
+	:cvar private _output_name: the list of output names to map.
+	:cvar private _neighbour_locator: data structure used internally to find
+		neighbors.
+	:cvar private _mode: indicate if new file will be interpolated using point
+		data or cell data.
+	:cvar private _n_neighbors: number of neighbors to find for interpolate new
+		value.
+	:cvar private _interpolate_func: function to interpolate new value.
+	"""
+
+	def __init__(self):
+		self._output_name = []
+		self._neighbour_locator = None
+		self._mode = "point"
+		self._n_neighbors = 1
+		self._interpolate_func = Mapper.default_interpolate_func
+
+	@staticmethod
+	def get_cell_centroid(vertices):
+		"""
+		This method compute centroid of a cell defined by `vertices`.
+		The centroid is compute as mean of all `vertices`
+
+		:param numpy.ndarray vertices: a n_vertices-by-3 where each row contain
+			coordinates of vertices
+		:return: centroid of cell
+		:rtype: numpy.ndarray
+		"""
+		assert (vertices.shape[1] == 3), "Not 3D vertices, something's wrong"
+		centroid = np.mean(vertices, axis=0)
+		return centroid
+
+	@staticmethod
+	def default_interpolate_func(values, distance):
+		if not distance.all():
+			return values[np.where(distance == 0)]
+
+		return np.average(values, weights=np.reciprocal(distance), axis=0)
+
+	@property
+	def output_name(self):
+		"""
+		The output names to map.
+
+		:getter: Returns the output names
+		:setter: Sets the output names
+		:type: list
+		"""
+		return self._output_name
+
+	@output_name.setter
+	def output_name(self, output_name):
+
+		if not isinstance(output_name, list):
+			output_name = [output_name]
+
+		for name in output_name:
+			assert (isinstance(name, str))
+
+		self._output_name = output_name
+
+	@property
+	def interpolate_function(self):
+		"""
+		The function to interpolate value of a new point using neighbors value.
+		It has to take as arguments two `numpy.array`: the first indicates
+		neighbors value (one row for each value) and the second indicates
+		distance from interpolated point.
```

Comment:
*the distance from the interpolated point.
Please provide a snippet of code as example.